pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -832,7 +853,12 @@ public void replaceConsumingSegment(String segmentName)
     File indexDir = new File(_indexDir, segmentName);
     // Get a new index loading config with latest table config and schema to load the segment
     IndexLoadingConfig indexLoadingConfig = fetchIndexLoadingConfig();
-    addSegment(ImmutableSegmentLoader.load(indexDir, indexLoadingConfig, _segmentOperationsThrottler));
+    ImmutableSegment immutableSegment =
+        ImmutableSegmentLoader.load(indexDir, indexLoadingConfig, _segmentOperationsThrottler);
+
+    // Fetch ZK metadata to pass to addSegment so that ZK creation time is set automatically
+    SegmentZKMetadata zkMetadata = fetchZKMetadataNullable(segmentName);
```

Comment:
`replaceConsumingSegment` is only called in RSDM, which has a member var with SegmentZKMetadata.

Fetching it here adds an extra ZK call which is avoidable.